### PR TITLE
Site Creation: Site Info VM, Fragment and Unit Tests

### LINKED ISSUE #8503: [Site Creation] Step: Info UI
Modify/Create UI for Info step in the Site Creation flow.

More info #8534

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #8503, #8504, #8535. This PR adds the fragment, the view model and the unit tests for the new site info screen. The screen itself is pretty basic and the only communication necessary from fragment to vm is the update to `businessName` and the `tagLine` and the only communication back is the state of the `skip/next` button.

The PR follows the same patterns used by the verticals and segments screens created by @malinajirka. It does not connect to the main VM yet which will be done in a separate PR depending on how we want to handle the loading.

Here are the screenshots for the screen in empty and filled states in both portrait and landscape orientations for both a phone and a tablet: https://cloudup.com/c-jAAw9KjmW @SylvesterWilmott If I missed any screenshots, let me know and I can add it for you.

To test:
* The `NEW_SITE_CREATION_ENABLED` flag needs to be enabled in `WordPress/build.gradle`.
* Since the fragment is not added to the flow yet, it'll need to be done to be able to test. One easy way to do that would be to change the `showFragment` in `NewSiteCreationActivity.onCreate()` to instead show the newly added fragment. Once that's done, you can go to the site selection screen, tap on the `+` icon and select "Create WordPress.com site" to see it.
* Make sure that the `Skip` button is visible initially and whenever both the business name and the tag line is empty
* Make sure the `Next` button is visible whenever either the business name or the tag line is filled
* Make sure the position of the `Next/Skip` button is on the bottom right of the page and pushed during keyboard transitions. When the button is not visible in the screen, make sure you can scroll to it and the position makes sense.
* Remember to rotate your screen once in a while.

### CODE REVIEW

File: WordPress/src/test/java/org/wordpress/android/ui/sitecreation/siteinfo/NewSiteCreationSiteInfoViewModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+    @Before
+    fun setUp() {
+        viewModel = NewSiteCreationSiteInfoViewModel()
+        viewModel.uiState.observeForever(uiStateObserver)
+    }
+
+    @Test
+    fun verifyInitialUiStateIsEmpty() {
+        assertThat(viewModel.uiState.value).isEqualToComparingFieldByField(EMPTY_UI_STATE)
```
Comment (malinajirka) [inline WordPress/src/test/java/org/wordpress/android/ui/sitecreation/siteinfo/NewSiteCreationSiteInfoViewModelTest.kt]: TIL isEqualToComparingFieldByField :)

**Comment by malinajirka** [comment]
Thanks Oguz! The code and the behaviour looks great! 

I've noticed three differences in design
- The title doesn't use Roboto-medium (It might be worth creating an issue as I haven't used it on any of the screens)
- The bottom line of a non-fucused editext is too dark (But I assume we use an app-wide style, right?)
- The right border(shadow) of the skip button is cut off
screenshot_1543914596
But let's wait for @SylvesterWilmott

**Comment by malinajirka** [comment]
Thanks @oguzkocer !

>>The title doesn't use Roboto-medium (It might be worth creating an issue as I haven't used it on any of the screens)
>Afaik, we are always using the same fonts to keep things consistent, so I didn't actually check if they were different. @SylvesterWilmott would you like me to change the font to Roboto-medium?

I think we are using `sans-serif-light` and `sans-serif-medium` (which are actually Roboto-light and Roboto-medium) at a couple of places in the app. And I actually used them even on the segments and verticals screen I just forgot about it:P.

>Great catch! I don't think we have a common style for this. I've added the tint color in da1bd5c.

👍 Thanks!

>>The right border(shadow) of the skip button is cut off
>As far as I know there is nothing that'd block the shadow. Are you sure there is supposed to be a shadow on the right side for our regular buttons? If so, do you mind checking the code, maybe I am missing something?

I tried to put an `End` margin before I wrote that comment and it fixed it. So I'm sure that there is something wrong, but I'm not sure what yet. I'll take a look later in the afternoon.

>I've just added a maxWidth for the necessary layouts since the values are large enough that it won't affect phones.

We may want to consider putting it into `site_creation_activity.xml` or even better `site_creation_form_screen.xml` (Unless we plan to remove the BaseFragment, which is worth considering). If you decide to edit it, can you please remove `site_creation_screen_horizontal_padding` and it's usages (SegmentsScreen, VerticalsScreen). Thanks!


>I am also not sure if I understood what you mean by "We can then give it a shadow when something passes beneath it". Do you mean that the shadow would only be visible if there is content behind it, but would be invisible otherwise? If so, I don't think we can tell that in runtime, but I am not sure. I could possibly show it while the keyboard is visible, if that works.

I agree with you that taking up the space isn't the best idea - we could theoretically use a `Done` button in the toolbar maybe?
Regarding the shadow -> I remember there is similar functionality on the `Login epilogue` screen. So if we decide to use it I think we can copy it from there.



This screen already works as expected and there are just a few minor design issues. If you want, feel free to merge it and create a separate PR with

**Comment by malinajirka** [comment]
>The right border(shadow) of the skip button is cut off

I'm not sure why, but ConstraintLayout apparently doesn't get along with nested LinearLayout. Probably as its main purpose is to flatten the hierarchy it's not expected to be used with nested layouts. The shadow seems ok when I removed the LinearLayout.

SideNotes - I'm sorry I haven't catch this during the review - I should have looked into the AS design preview.
1. I'm a ConstraintLayout newbie, however I always felt like all the views are required to have both horizontal and vertical constraints. Tbh I'm surprised the AS is not complaining that some of the views are missing horizontal constraints. 
2. I think we shouldn't use `match_parent` within the constraintLayout
https://developer.android.com/reference/android/support/constraint/ConstraintLayout
```
Important: MATCH_PARENT is not recommended for widgets contained in a ConstraintLayout. Similar behaviour can be defined by using MATCH_CONSTRAINT with the corresponding left/right or top/bottom constraints being set to "parent".
```

**Comment by SylvesterWilmott** [comment]
>I can fix it to bottom, but it'd block the content especially on the smaller devices. The keyboard already takes up a lot of space, I am not a big fan of blocking even more and making it harder to navigate the text fields. Even in a large device like Pixel 2, it looks like this without the border: https://cloudup.com/cDF0LHkHQEt.

It's really the lesser of two evils here, the button should be reachable without dismissing the keyboard.

>Do you mean that the shadow would only be visible if there is content behind it, but would be invisible otherwise? If so, I don't think we can tell that in runtime, but I am not sure. I could possibly show it while the keyboard is visible, if that works.

Yeah, that's correct. Can we show the shadow if the view is scrollable? I think this would then be a good solution for the other steps in the flow and other orientations.

Also, to clarify, we only want to hide/show the shadow and not the background of the bar.

**Comment by malinajirka** [comment]
Thanks for the changes @oguzkocer!

>@malinajirka I am not sure which LinearLayout you're referring to here. We only had one LinearLayout in that screen which was used for headers and I really don't think it has any effect on the clipping of the shadow of the button. 

Yep, that's why I mentioned I'm not sure why it's causing the issue. However, removing the HeaderLayout(LinearLayout) fixed the issue for me and I didn't have to add `clipChildren` and `clipToPadding`. Nevertheless it doesn't really matter, I'm glad it works now + it's explicit. Thanks! ;)


Everything LGTM. I'm merging this since all the "open" discussions have their own tickets.

P.S. I agree with Oguz regarding the button.

**Comment by SylvesterWilmott** [comment]
>The button is currently reachable while the keyboard is visible by scrolling, so I don't think it's that big of a deal. In fact, when we make this change, some users will still need to scroll if they have a smaller device, but this time it'll be to reach the second text field. The trade-off here is that, in one solution the button is reachable with a scroll (which btw we could potentially do when the user is editing the second text field) and in the other the large devices will show everything with the skip button background slightly blocking the hint and in smaller devices the user still needs to scroll to reach the second text field while also having less real estate to work with.

I hear you but there are a few reasons to fix the button to the top of the keyboard:

- The second text field (the one that will be partially covered on small devices) is an optional field, this puts it at a lower priority than a button that will be used to proceed to the next step which is a high priority action
- We need to use a scalable solution that works for any layout, if we add more fields or options to this screen we can end up pushing the button off the bottom edge of the screen. This is true for any of the other steps as well
- By fixing it in place, the button to proceed/skip to the next step appears in the same place each time
- It may not be immediately clear that a button exists below the fold/keyboard, however placing a button on a bar with a shadow implies visually that something is beneath it
- This conversation already occurred during the sign up/log in design process and the design pattern of having a fixed button on top of the keyboard was the accepted solution

I agree it's not ideal but it's the nature of trade offs for small devices.

**Comment by SylvesterWilmott** [comment]
Just to clarify as well @oguzkocer , the first text field should be focused automatically in this view.

**Comment by oguzkocer** [comment]
> I hear you but there are a few reasons to fix the button to the top of the keyboard:

Thanks for the detailed explanation, that makes a lot of sense and that's why I rarely question design decisions. I am glad that I did though because this does put things into perspective for me.

> the first text field should be focused automatically in this view.

I added a note for this in #8736.

**Pull request merged**